packages/cli/src/util/env/get-env-records.ts

```diff
@@ -49,3 +49,41 @@ export default async function getEnvRecords(
 
   return client.fetch<{ envs: ProjectEnvVariable[] }>(url);
 }
+
+export async function pullEnvRecords(
+  output: Output,
+  client: Client,
+  projectId: string,
+  source: EnvRecordsSource,
+  {
+    target,
+    gitBranch,
+  }: {
+    target?: ProjectEnvTarget | string;
+    gitBranch?: string;
+  } = {}
+) {
+  output.debug(
+    `Fetching Environment Variables of project ${projectId} and target ${target}`
+  );
+  const query = new URLSearchParams();
+
+  let url = `/v1/env/pull/${projectId}`;
+
+  if (target) {
+    const gitBranchPath = gitBranch ? `/${gitBranch}` : '';
+    url += `/${target}${gitBranchPath}`;
+  }
+
+  if (source) {
+    query.set('source', source);
+  }
+  if (Array.from(query).length) {
```

Comment:
nit: don't rely on implicit type conversion. add a conditional
```suggestion
  if (Array.from(query).length > 0) {
```